include/tc/core/cuda/cuda_tc_executor.h

```diff
@@ -23,20 +28,18 @@
 #include "tc/core/utils/dlpack.h"
 #include "tc/lang/parser.h"
 
-#include <dlpack/dlpack.h>
-
 namespace tc {
 
 class CudaTcExecutor : public ::tc::TcExecutor {
  public:
   CudaTcExecutor(
-      const std::string& def,
-      const std::vector<const DLTensor*>& inputsInfo)
-      : TcExecutor(def, inputsInfo) {}
-  CudaTcExecutor(
-      lang::TreeRef tree,
-      const std::vector<const DLTensor*>& inputsInfo)
-      : TcExecutor(tree, inputsInfo) {}
+      std::string id,
+      std::vector<const DLTensor*> inputsInfo,
+      const std::string& options,
+      lang::TreeRef tcDefinition,
+      size_t handle)
+      : TcExecutor(id, inputsInfo, options, tcDefinition, handle) {}
```

Comment:
why do we need to pass the options, handle and tcdef as well? isn't handle obtained for the `tcdef` itself?

Reply:
I think the handle itself is redundant here. It's the index of executor in the list...

Reply:
agreed, would be nice to see this fixed :)

Reply:
is this getting fixed?

Reply:
need to dig deeper, for now this just consolidates the constructors and kills some code but doesn't yet change the behavior. Prob a followup commit